index.html

```diff
@@ -4318,24 +4285,26 @@ <h2>Metadata Structure</h2>
 set</a>, <a data-cite="INFRA#boolean">boolean</a>, or
 <a data-cite="INFRA#null">null</a>. The values within any complex data
 structures such as maps and lists MUST be one of these data types as well.
-All metadata property definitions MUST define the value type, including any
+All metadata property definitions registered in the DID Specification
+Registries [[?DID-SPEC-REGISTRIES]] MUST define the value type, including any
 additional formats or restrictions to that value (for example, a string
 formatted as a date or as a decimal integer). It is RECOMMENDED that property
-definitions use strings for values.
+definitions use strings for values. The entire metadata structure MUST be
+serializable in a conforming <a>representation</a>.
```

Comment:
Why was this additional requirement added? The implementation might do any number of things to serialize the metadata separately from the document itself. As long as it gets in and out of the map, that's fine. By requiring the metadata to re-use the type and data structure system it inherits this property anyway.

```suggestion
definitions use strings for values.
```

Reply:
Yes, the implementation might do any number of things -- but we can't test "any number of things" -- we need to test a "specific thing". The text was added to provide something specific we can test for... it doesn't say you have to serialize it... it says it "must be serializable" -- that is, you must be able to do it (and the test suite will test that). Implementations aren't required to serialize, just ensure that it is possible to do so.

Reply:
I do not agree that this makes it any more "testable" than before. The burden is on the test suite to serialize the result in either case, and a conformant DID document already needs to be representable. This requirement is redundant and implies that there are DID documents that could possibly :not: be representable in some form.

Reply:
There is no such thing as "a conformant representation" for metadata. The representations are defined in terms of the did documents. This requirement does not make sense.

Reply:
I thought we wanted to be able to return metadata as cbor, json, yaml, etc... are we saying we want to be able to return a json did document with cbor meta data?

My understanding was that the entire response values would be in the requested representation... so if you asked for json, you would not get back cbor... but perhaps you see this as way to return response data in http codes / headers... which will never be JSON / CBOR?

Reply:
Metadata is not ever defined in terms of JSON, CBOR, or anything. It's a data structure (a map). 

The ability to call the `resolve` function for a specific platform is going to require some way to get the map values in and out. We specify this in terms of the data structure -- map -- and not in terms of a serialization for that data structure. The method of passing that data structure in and out is the binding of the abstract function to a specific implementation.

Reply:
It sounds like what you're writing here is a requirement for hooking functions to the test harness, and not requirements for the function itself. This is a dangerous line to cross.

Reply:
> There is no such thing as "a conformant representation" for metadata. 
> Metadata is not ever defined in terms of JSON, CBOR, or anything. It's a data structure (a map).

Here's the problem that we continue to have with the Resolution section:

The original authors intend an unspecified subset of it to be a purely abstract thing while simultaneously wanting there to be normative language related to the abstract concepts. While it is true that we can write normative statements about abstract concepts, it has been asserted that doing so is less useful than being able to have testable statements. This point has been made repeatedly throughout the past decade or more at W3C, across multiple Working Groups, with repeated consensus positions reaffirming that groups shouldn't write normative language around abstract concepts that are not realizable in concrete tests.

In other words, we cannot test abstract things in the test suite if they are never realized as something concrete. It is problematic to have normative language on abstract concepts because it's very easy for people to make subjective assertions about their conformance to those statements. When you don't make things testable, you harm interoperability.

We have three options that were explored:

1. Migrate the entire Resolution section into a NOTE.
2. Remove normative language that is associated with abstract concepts/statements. 
3. Make the normative language around abstract concepts/statements testable by providing a path to a concrete expression of those abstract concepts/statements.

The third item seemed to be the one that had the best chances of achieving consensus.

In order for that to happen, we need a proposal that enables us to get the metadata into a concrete form so the test suite can test it. One approach is to say that "the metadata structure MUST be serializable in a conforming representation". If we look at the JSON production rules, it doesn't say anything about production ONLY applying to DID Documents... it just states that "if you have data type X, then you can serialize by doing Y." ... and if we can do that with the metadata structure, then we have a path to testability.

I'm also open to any other paths to testability.

At this point, @jricher, we need a counter-proposal on how we get the metadata structure to testability, or we need you to clarify if you are objecting or if you can live with the language.

Reply:
I am not going to formally object to anything in here, but I am on record in saying that this is a bad idea that makes for a weak specification.

Reply:
In lieu of a counter-proposal that can achieve stronger consensus, I'm hearing that you won't object to keeping the language as it stands.